### PR TITLE
Updates the chef provisioner to allow specifying a channel

### PR DESCRIPTION
**Pull request by scotthain**
This also updates the omnitruck url to the current url.

Signed-off-by: Scott Hain <shain@chef.io>

### CODE REVIEW

File: builtin/provisioners/chef/linux_provisioner_test.go [hunk 1]
```diff
@@ -120,7 +120,26 @@ func TestResourceProvider_linuxInstallChefClient(t *testing.T) {
 
 			Commands: map[string]bool{
 				"curl -LO https://www.chef.io/chef/install.sh": true,
```
Comment (schisamo) [inline builtin/provisioners/chef/linux_provisioner_test.go]: We should update this URL to be `https://omnitruck.chef.io/install.sh` (along with all other spots `www.chef.io` is being used in test and no-test files).

File: builtin/provisioners/chef/linux_provisioner.go [hunk 2]
```diff
@@ -11,7 +11,7 @@ import (
 
 const (
 	chmod      = "find %s -maxdepth 1 -type f -exec /bin/chmod %d {} +"
-	installURL = "https://www.chef.io/chef/install.sh"
+	installURL = "https://omnitruck.chef.io/chef/install.sh"
```
Comment (schisamo) [inline builtin/provisioners/chef/linux_provisioner.go]: Drop the `chef` in this URL...we just want to use `https://omnitruck.chef.io/install.sh`, more details at https://docs.chef.io/packages.html#omnitruck-install-script-options

**Comment by scotthain** [comment]
Hey folks - any chance we could get an ETA on merging this? Thanks in advance!

**Comment by svanharmelen** [comment]
Thanks for this one @scotthain! LGTM and it also fixes PR #17020, so here goes...

**Pull request merged**